config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml

```diff
@@ -412,7 +412,6 @@ presubmits:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: capz-pr-upstream-k8s-ci-windows-containerd-serial-slow-main
   - name: pull-cluster-api-provider-azure-apidiff
-    cluster: eks-prow-build-cluster
```

Comment:
@mboersma @aramase I wonder if this apidiff job really needs the service account (see @rjsadow's comment below about the fact that our v1beta1 release branch test isn't using it).

Anyone know what that service account is there for and the likelihood that it'd be used in a `go-apidiff` flow? (seems unlikely to me)

Reply:
The service account was added in https://github.com/kubernetes/test-infra/pull/27188. @willie-yao Any thoughts on this?

Reply:
found this: https://github.com/kubernetes/test-infra/tree/master/workload-identity#migration-steps-when-not-interacting-with-gcp. Maybe it is required?

Reply:
Yeah, this makes more sense now.

Is this a potential gap for the new EKS prow pool? Should we replicate this service account so folks can continue to use workload-identity?

Reply:
Although the v1beta1 job (which doesn't have this config) doesn't seem to be having trouble uploading artifacts. Ref:

- https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3658/pull-cluster-api-provider-azure-apidiff-v1beta1/1671896480064999424/

Compare to main presubmit which has the workload-identity config:

- https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3667/pull-cluster-api-provider-azure-apidiff/1673385633856360448/